src/mlpack/core/optimizers/scd/descent_policies/cyclic_descent.hpp

```diff
@@ -0,0 +1,50 @@
+/**
+ * @file cyclic_descent.hpp
+ * @author Shikhar Bhardwaj
+ *
+ * Cyclic descent policy for Stochastic Co ordinate Descent (SCD).
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_SCD_DESCENT_POLICIES_CYCLIC_HPP
+#define MLPACK_CORE_OPTIMIZERS_SCD_DESCENT_POLICIES_CYCLIC_HPP
+
+#include <mlpack/core.hpp>
+
+namespace mlpack {
+namespace optimization {
+
+/**
+ * Cyclic descent policy for Stochastic Co-ordinate Descent(SCD). This
+ * descent scheme picks a the co-ordinate for the descent in a cyclic manner
+ * serially.
```

Comment:
Is there an easy paper to cite here for SCD?  It's always nice to provide references to the original paper.